nmdc_automation/re_iding/base.py

```diff
@@ -581,3 +584,154 @@ def _make_new_filename(self, new_activity_id: str,
+    """
+    updated_omics_processing = deepcopy(omics_processing)
+    # Check if we need to update the omics processing ID and add the legacy ID to the alternate identifiers
+    if not updated_omics_processing.id.startswith("nmdc:omprc-"):
+        updated_omics_processing = _update_omics_processing_alternative_identifiers(updated_omics_processing, updated_omics_processing.id)
+        new_omics_processing_id = _get_new_omics_processing_id(updated_omics_processing, api_client, identifiers_map)
+        updated_omics_processing.id = new_omics_processing_id
+
+    # Update the has_input array
+    has_input = updated_omics_processing.has_input
+    # Only Biosamples (nmdc;bsm-*) or ProcessedSamples (nmdc:prcsm-*) should be in the has_input array
+    has_input = [id for id in has_input if id.startswith("nmdc:bsm-") or id.startswith("nmdc:prcsm-")]
+    if nmdc_biosample_id not in has_input:
+        has_input.append(nmdc_biosample_id)
+        updated_omics_processing.has_input = has_input
+        logging.info(f"Added new biosample ID to has_input: {nmdc_biosample_id}")
+
+    # Update the part_of array with the new study ID
+    part_of = updated_omics_processing.part_of
+    # Remove gold: IDs from part_of array
+    part_of = [id for id in part_of if not id.startswith("gold:")]
+    if nmdc_study_id not in part_of:
+        part_of.append(nmdc_study_id)
+        updated_omics_processing.part_of = part_of
+        logging.info(f"Added new study ID to part_of: {nmdc_study_id}")
+
+    return updated_omics_processing
+
+
+def _get_omics_processing_legacy_id(omics_processing: nmdc.OmicsProcessing) -> str:
+    if not omics_processing.id.startswith("nmdc:omprc-"):
+        return omics_processing.id
+    elif omics_processing.gold_sequencing_project_identifiers:
+        return omics_processing.gold_sequencing_project_identifiers[0]
+    elif omics_processing.alternative_identifiers:
+        return omics_processing.alternative_identifiers[0]
+    else:
+        return omics_processing.id
+
+
+def _update_biosample_alternate_identifiers(biosample: nmdc.Biosample, legacy_biosample_id: str) -> nmdc.Biosample:
+    """
+    Update the appropriate alt identifiers slot depending on the Biosample legacy ID:
+    - gold_biosample_identifiers for legacy IDs starting with 'gold:'
+    - igsn_biosample_identifiers for legacy IDs starting with 'igsn:'
+    - emsl_biosample_identifiers for legacy IDs starting with 'emsl:'
+    """
+    if legacy_biosample_id.startswith("gold:"):
+        if legacy_biosample_id not in biosample.gold_biosample_identifiers:
+            biosample.gold_biosample_identifiers.append(legacy_biosample_id)
+            logging.info(f"Added legacy biosample ID to gold_biosample_identifiers: {legacy_biosample_id}")
+    elif legacy_biosample_id.startswith("igsn:"):
+        if legacy_biosample_id not in biosample.igsn_biosample_identifiers:
+            biosample.igsn_biosample_identifiers.append(legacy_biosample_id)
```

Comment:
review append vs extend, I think we want extend here to add an existing list vs adding a new list.

https://www.freecodecamp.org/news/append-in-python-how-to-append-to-a-list-or-an-array/#:~:text=append()%20adds%20the%20new,extend()%20method.